theano/tensor/nnet/tests/test_abstract_conv.py

```diff
@@ -1931,3 +1931,129 @@ def test_interface(self):
         fun = theano.function([x_sym, dfilter_sym, pfilter_sym], sep_op, mode='FAST_RUN')
         top = fun(x[:, :, :3, :3], depthwise_filter, pointwise_filter)
         utt.assert_allclose(top, precomp_output)
+
+
+class TestUnsharedConv(unittest.TestCase):
+    conv2d = theano.tensor.nnet.abstract_conv.AbstractConv2d
+    conv2d_gradw = theano.tensor.nnet.abstract_conv.AbstractConv2d_gradWeights
+    conv2d_gradi = theano.tensor.nnet.abstract_conv.AbstractConv2d_gradInputs
+    conv2d_op = theano.tensor.nnet.abstract_conv.AbstractConv2d
+    conv2d_gradw_op = theano.tensor.nnet.abstract_conv.AbstractConv2d_gradWeights
+    conv2d_gradi_op = theano.tensor.nnet.abstract_conv.AbstractConv2d_gradInputs
+
+    def setUp(self):
+        self.mode = theano.compile.mode.Mode(optimizer='None')
+
+        self.img_shape = [(2, 6, 5, 5), (1, 4, 7, 5), (1, 8, 5, 3), (1, 4, 7, 7)]
+        self.kern_shape = [(2, 3, 3, 6, 3, 3), (2, 6, 4, 4, 5, 3), (4, 2, 1, 2, 3, 3), (4, 4, 5, 2, 5, 3)]
+        self.topgrad_shape = [(2, 2, 3, 3), (1, 2, 6, 4), (1, 4, 2, 1), (1, 4, 4, 5)]
```

Comment:
Can you make these any smaller? (Or remove a few?) The `verify_grad` on these large inputs is quite slow (see the [Travis log](https://travis-ci.org/Theano/Theano/jobs/264669314#L3204-L3206)), so reducing the size of the inputs would help.

Alternative: run `verify_grad` for a subset of test cases (for example, pass a list of booleans in your `zip` loop to say when you want to run it). The rest of the test isn't too slow.